tests/minikube/setup.sh

```diff
@@ -139,70 +196,108 @@ github_asset_binary() {
-      # are present in PATH.
-      nix_shell_prerequisites=("conntrack" "minikube" "crictl" "curl" "jq" "awk" "systemctl" "docker")
-      for bin in "${nix_shell_prerequisites[@]}"; do
+      # Check if nix-shell prerequisites are present in PATH.
+      for bin in "${NIX_SHELL_PREREQUISITES[@]}"; do
         must_exists_in_path "${bin}" >/dev/null
       done
 
       echo "Installing prerequisites for $os-$arch..."
 
       # Install cri-dockerd, if not installed.
-      cri_dockerd_error=""
-      command -v cri-dockerd >/dev/null || cri_dockerd_error=$?
+      local cri_dockerd_error
+      which cri-dockerd &>/dev/null || cri_dockerd_error=$?
       if [ -n "${cri_dockerd_error}" ]; then
         echo "Downloading latest version of cri-dockerd..."
-        url=$(github_asset_url_from_latest_release "Mirantis" "cri-dockerd" "^(cri-dockerd-[0-9]+.[0-9]+.[0-9]+.amd64.tgz)$")
+
+        local -r url=$(github_asset_url_from_latest_release \
+          "Mirantis" "cri-dockerd" "^(cri-dockerd-[0-9]+.[0-9]+.[0-9]+.amd64.tgz)$")
         github_asset_binary "${url}" "${workspace}" "cri-dockerd.tgz"
         tar -xf "${workspace}"/cri-dockerd.tgz -C "${workspace}"
         mkdir -p /usr/local/bin
-        install -o root -g root -m 0755 "${workspace}"/cri-dockerd/cri-dockerd /usr/local/bin/cri-dockerd
-        command -v cri-dockerd >/dev/null || export PATH=$PATH:/usr/local/bin
+        local -r bin_dir="/usr/local/bin"
+        install -o root -g root -m 0755 "${workspace}"/cri-dockerd/cri-dockerd "$bin_dir"
+
+        local original_path
+        which cri-dockerd &>/dev/null \
+          || original_path=$PATH && export PATH=$PATH:/usr/local/bin
         must_exists_in_path "cri-dockerd" "failed to install cri-dockerd" >/dev/null
+
+        # Cleanup entry for cri-dockerd binary.
+        local cleanup_script="# Removes the cri-dockerd binary
+rm -f ${bin_dir}/cri-dockerd"
+        if [ -n "$original_path" ]; then
+          cleanup_script="$cleanup_script
+# Reinstates the previous value for PATH
+export PATH=$original_path"
+          unset original_path
+        fi
+        prepend_cleanup_script "$cleanup_script"
+
         echo "Downloaded latest version of cri-dockerd."
       fi
       # Check if cri-docker.socket systemd service is active. While this is extremely
       # unlikely if cri-dockerd wasn't already installed, it is possible.
-      cri_dockerd_service_error=""
+      local cri_dockerd_service_error
       systemctl is-active --quiet cri-docker.socket || cri_dockerd_service_error=$?
       if [ -n "${cri_dockerd_service_error}" ]; then
         # Download systemd service files
-        cri_dockerd_version="v$(cri-dockerd --version 2>&1 | awk '{print $2}')"
-        systemd_service_files_url="https://raw.githubusercontent.com/Mirantis/cri-dockerd/${cri_dockerd_version}/packaging/systemd"
-        cd ${workspace}
+        local -r cri_dockerd_version="v$(cri-dockerd --version 2>&1 | awk '{print $2}')"
```

Comment:
Wondering if we're doing too much here, and simply set pre-requisite for user to set this up before hand..

but anyways, you might want to check first if the system is using systemd (might not be, depending on the linux distro)

Reply:
Added a check to see if 'systemctl' exists, which is a good sign that it's a systemd Linux system.
>Wondering if we're doing too much here, and simply set pre-requisite for user to set this up before hand
This might come in handy for CI runs with this script. I've made the installations enabled by default. Also, added a `-s` flag option to skip prerequiste installation.

I plan to add in a kvm-based profile for a personal-use scenario. Keep this in TODO, for now.

Reply:
> Added a check to see if 'systemctl' exists, which is a good sign that it's a systemd Linux system.

Removed it. There is already a check for this, I'd forgotten about it:
```
NIX_SHELL_PREREQUISITES=("conntrack" "minikube" "crictl" "curl" "jq" "yq" "awk" "systemctl" "docker")
```
```
      for bin in "${NIX_SHELL_PREREQUISITES[@]}"; do
        must_exists_in_path "${bin}" >/dev/null
      done
```